core/test/libraries/Positions.t.sol

```diff
@@ -19,6 +19,57 @@ contract PositionsTest is Test {
         positions[5] = 678;
     }
 
+    function test_memoryZip(int24 xl, int24 xu, int24 yl, int24 yu, int24 zl, int24 zu) public {
+        int24[6] memory arr = [xl, xu, yl, yu, zl, zu];
+        vm.expectSafeMemory(0x00, 0x60);
+        zip(arr);
+    }
+
+    function test_memoryExtract(int24 xl, int24 xu, int24 yl, int24 yu, int24 zl, int24 zu) public {
+        uint64 count = 0;
+        if (xl != xu) count += 1;
+        if (yl != yu) count += 1;
+        if (zl != zu) count += 1;
+        uint256 zipped = zip([xl, xu, yl, yu, zl, zu]);
+
+        uint64 ptr;
+        assembly ("memory-safe") {
+            ptr := mload(0x40)
+        }
+
+        vm.expectSafeMemory(ptr, ptr + 32 * (1 + 2 * count));
+        extract(zipped);
+    }
+
+    function test_memoryExtractDirty(int24 xl, int24 xu, uint256 dirt) public {
+        uint256 zipped = zip([xl, xu, xl, xu, xl, xu]);
+
+        // Write data after the free memory pointer, without updating the pointer
+        // (make it dirty)
+        assembly ("memory-safe") {
+            let ptr := mload(0x40)
+            mstore(ptr, dirt)
+            mstore(add(ptr, 0x20), dirt)
+            mstore(add(ptr, 0x40), dirt)
+            mstore(add(ptr, 0x60), dirt)
+            mstore(add(ptr, 0x80), dirt)
+            // ...could add more, doesn't really matter
+        }
+
+        int24[] memory extracted = extract(zipped);
```

Comment:
Goal is to make sure `extract` still works, even when the memory it's writing to is dirty (see this [big warning](https://docs.soliditylang.org/en/v0.8.19/internals/layout_in_memory.html))